src/code/z_play.c

```diff
@@ -15,6 +15,14 @@ u64 D_801614D0[0xA00];
 
 void Play_SpawnScene(PlayState* this, s32 sceneNum, s32 spawn);
 
+// If R_ENABLE_PLAY_LOGS is enabled, the number "1" with a file and line number will be printed during
+// various phases of the Play update cycle. This can be helpful for debugging at what stage a problem may be occuring.
+// Due to the high number of logs occuring rapidly, the game will run slow with this enabled.
```

Comment:
The slowdown is an implementation detail of the emulator / IS64 scripts imo (and especially the pj64 one is incredibly slow),

An easy way to test that is first 1) changing `#define gISVDbgPrnAdrs ((ISVDbg*)0xB3FF0000)` to use `0xB3FE0000`, making the game read/write to a region unrelated to the is64, then 2) removing the `if (data != ASCII_TO_U32('I', 'S', '6', '4')) {` magic check

This makes the game run exactly as if it was printing logs through the IS64, but it's using the wrong part of memory which makes the emulator not react to any of it.
And with this `R_ENABLE_PLAY_LOGS = true;` doesn't cause any slowdown

So printing doesn't really slow the game down, but it slows down emulation when the emulator supports the IS64 regs

Reply:
I did this one better and tested on console 

I am very confident I've seen a high abundance of logs slow down on console before, but this is apparently not enough, cause it runs fine

Reply:
Yeah of course printing logs isn't free but this isn't a lot at all
Fwiw the is64 emulator built into mupen can almost keep up with the play logs enabled